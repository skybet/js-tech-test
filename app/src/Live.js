import React, { useContext } from "react";
import Market from "./components/Market";
import useLiveEvents from "./hooks/useLiveEvents";
import { OddsContext } from "./App";

const Live = () => {
  const [oddsType, toggle] = useContext(OddsContext);
  const events = useLiveEvents();

  return events.length === 0 ? (
    <div>Loading...</div>
  ) : (
    <React.Fragment>
      <button onClick={toggle}>Odds: {oddsType}</button>
      {events.map(({ eventId, name, markets }) => (
        <div key={eventId} className="c-event">
          <h2 key={eventId}>{name}</h2>
          <Market id={markets[0]} />
        </div>
      ))}
    </React.Fragment>
  );
};

export default Live;
