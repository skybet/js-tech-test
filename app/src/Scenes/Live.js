import React from "react";
import Market from "../components/Market";
import OddsToggle from "../components/OddsToggle";
import { useLiveEvents } from "../hooks/";
import { Link } from "react-router-dom";

const Live = () => {
  const events = useLiveEvents();

  return events.length === 0 ? (
    <div>Loading...</div>
  ) : (
    <React.Fragment>
      <OddsToggle />
      {events.map(({ eventId, markets, competitors, scores }) => (
        <div key={eventId} className="c-event">
          <div key={eventId} className="c-event__name">
            <div className="c-event__competitors">
              {competitors[0].name}
              <br />
              {competitors[1].name}
              <br />
              <Link className="c-event__full-market" to={`/event/${eventId}`}>
                View Full Market
              </Link>
            </div>
            <div className="c-event__score">
              {scores.home}
              <br />
              {scores.away}
            </div>
          </div>
          <Market id={markets[0]} />
        </div>
      ))}
    </React.Fragment>
  );
};

export default Live;
