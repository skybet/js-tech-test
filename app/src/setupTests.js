// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { render } from "@testing-library/react";
import { OddsContext } from "./App";

jest.mock("./hooks/useMarket", () =>
  jest.fn(() => ({
    marketId: 111,
    name: "Both teams to score",
    outcomes: [111, 222]
  }))
);

jest.mock("./hooks/useOutcome", () =>
  jest.fn(() => ({
    name: "Yes",
    price: {
      num: 3,
      den: 1,
      decimal: 4
    }
  }))
);

jest.mock("./hooks/useLiveEvents", () =>
  jest.fn(() => [
    {
      eventId: 111,
      name: "Everton vs West Ham",
      markets: [111, 222],
      competitors: [{ name: "Everton" }, { name: "West Ham" }],
      scores: { home: 3, away: 0 }
    }
  ])
);

export const renderWithContext = (component, oddsType = "fractional") =>
  render(
    <OddsContext.Provider value={[oddsType, jest.fn()]}>
      {component}
    </OddsContext.Provider>
  );
