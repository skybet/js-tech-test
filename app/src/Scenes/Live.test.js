import React from "react";
import Live from "./Live";
import { renderWithContext } from "../setupTests";

describe("<Live />", () => {
  test("renders as expected", async () => {
    const { asFragment } = renderWithContext(<Live />);
    expect(asFragment()).toMatchSnapshot();
  });
});
