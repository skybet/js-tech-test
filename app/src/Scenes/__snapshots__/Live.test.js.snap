// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Live /> renders as expected 1`] = `
<DocumentFragment>
  <button
    class="c-button"
  >
    Odds: fractional
  </button>
  <div
    class="c-event"
  >
    <div
      class="c-event__name"
    >
      <div
        class="c-event__competitors"
      >
        Everton
        <br />
        West Ham
      </div>
      <div
        class="c-event__score"
      >
        3
        <br />
        0
      </div>
    </div>
    <div
      class="c-market"
    >
      <div
        class="c-market__title"
      >
        Both teams to score
      </div>
      <div>
        <div
          class="c-outcome"
        >
          <span
            class="u-clr--blue u-mr--100"
          >
            Yes
          </span>
          <span
            class="u-clr--red"
          >
            3/1
          </span>
        </div>
        <div
          class="c-outcome"
        >
          <span
            class="u-clr--blue u-mr--100"
          >
            Yes
          </span>
          <span
            class="u-clr--red"
          >
            3/1
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
